src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/SolutionService.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright(c) Microsoft.All Rights Reserved.Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    /// <inheritdoc cref="ISolutionService"/>
+    [Export(typeof(ISolutionService))]
+    internal sealed class SolutionService : ISolutionService, IVsSolutionEvents, IVsPrioritizedSolutionEvents, IDisposable
+    {
+        private const int StateUninitialized = 0;
+        private const int StateListening = 1;
+        private const int StateDisposed = 2;
+
+        private readonly IVsUIService<IVsSolution> _solution;
+        private readonly JoinableTaskContext _joinableTaskContext;
+
+        private int _state = StateUninitialized;
+        private uint _cookie = VSConstants.VSCOOKIE_NIL;
+        
+        /// <inheritdoc />
+        public bool IsSolutionClosing { get; private set; }
```

Comment:
Nit, this is a property and hence under constructor.

Reply:
Is that a standard everywhere? Personally I put stateful properties above the ctor (along with fields and other state), and computed properties after the ctor.

Reply:
Yes.